OpenRA.Game/Traits/World/ActorMap.cs

```diff
@@ -47,12 +47,12 @@ class CellTrigger
 			public readonly CPos[] Footprint;
 			public bool Dirty;
 
-			readonly Action<Actor> onActorEntered;
-			readonly Action<Actor> onActorExited;
+			readonly Action<Tuple<Actor, Player>> onActorEntered;
+			readonly Action<Tuple<Actor, Player>> onActorExited;
```

Comment:
I'm currently in the middle of an update of this and I realized - you have to start with `OnActorChanged`. While doing that does trigger in your "makes a player lose if an enemy unit crosses a trigger region. An enemy captures one of your units inside the region during the same tick that that is killed" example, having `OnExited` running before `OnActorChanged` would crash the game if a player hijacks an enemy tank which was just leaving his Cloak Generator's range in TS (since the Mutant Hijacker is also a Nod unit).